packages/server/src/db/index.js

```diff
@@ -704,9 +702,76 @@ async function getGrants({
                     queryBuilder.orderBy(orderArgs[0], orderDirection);
                 }
             }
+            queryBuilder.limit(perPage);
+            queryBuilder.offset((currentPage - 1) * perPage);
+        });
+
+    const counts = await knex(TABLES.grants)
```

Comment:
duplicating the same query as above instead of abstracting it to a separate function. We will be discarding this entire code-path in a few weeks so I felt this solution was adequate for the time being.